interfaces/builtin/serial_port.go

```diff
@@ -43,7 +43,13 @@ func (iface *SerialPortInterface) String() string {
 
 // Pattern to match allowed serial device nodes, path attributes will be
 // compared to this for validity when not using udev identification
-var serialDeviceNodePattern = regexp.MustCompile("^/dev/tty[A-Z]{1,3}[0-9]{1,3}$")
+// Known device node patterns we need to support
+//  - ttyUSBX  (UART over USB devices)
+//  - ttyACMX  (ACM modem devices )
+//  - ttyXRUSBx  (Exar Corp. USB UART devices)
+//  - ttySX (UART serial ports)
+//  - ttyOX (UART serial ports on ARM)
+var serialDeviceNodePattern = regexp.MustCompile("^/dev/tty[A-Z]+[0-9]+$")
```

Comment:
there is a gazillion of possible names ... ttyO[0-9) is originally for OMAP while TI uses a bunch of other names as well for other chips, every vendor makes up their own device name here (there is also ttyMSM which isnt even in the list above). while more precise means more control it also means we'll end up with a massive list over time, do we want that maintenance burden ?